docs/enhancements/fep-001-api-fleet-templating.md

```diff
@@ -0,0 +1,303 @@
+  evaluation changes) the device controller will queue the device to be re-evaluated and the
+  `TemplateVersion` re-rendered.
+
+#### Expected behavior on issues
+
+##### Device cannot be rendered from Template due to missing label
+When a device cannot be rendered for some reason (e.g. missing label), the fleet-controller
+will mark the device with a label `fleet-controller/failed-to-reconcile=true`, and add an annotation
+`fleet-controller/failed-to-reconcile-reason` with a human-readable reason. The fleet will include
+a status condition `DeviceFailedToReconcile`.
+
+Once all devices are corrected and can be reconciled again this status condition will be removed.
+
+#### Device cannot generate /device/{name}/rendered version due to missing resource
+When a device cannot render the template due to missing resources (e.g. git repo not found, secret not found),
+the device-controller will set a status condition on the device `MissingResource` with a human-readable reason.
+
+### User Stories 
+
+#### Story 1: network access credentials
+As an IT administrator, I want to use site-specific network access credentials for devices in the same fleet, so that I do not have to deal with the complexity of maintaining one fleet per site.
+
+#### Story 2: site specific endpoints
+As an IT administrator, I want to provide site-specific server endpoints to devices that depend on LAN-configured edge servers.
+
+#### Story 3: site specific certificate bundles
+As an IT administrator, I want to provide site-specific certificate bundles to enable trust with the local site infrastructure.
+
+#### Story 4: injecting the device name into the configuration
+As an IT administrator, I want the name of my MicroShift instance in ACM to have the same cluster name as the underlying device's name or fingerprint, so I can easily correlate between the two.
+
+#### Story 5: site specific telemetry endpoints
+As an IT administrator, I want to provide site-specific/region-specific telemetry endpoints to devices.
+
+#### Story 6: dialing up verbosity for logging in my user application in specific devices
+As an IT administrator, I want to dial up the verbosity for logging on a specific device, so I can troubleshoot it.
+
+#### Story 7: region specific application service endpoints
+As an IT administrator, I want to provide region-specific service endpoints to my applications
+
+#### Story 8: region specific radio bands/regulations
+As a developer, I want to ensure that my devices automatically comply with local regulations when enrolled in specific regions. (i.e. configured with region-specific radio bands for WiFi, 5G, or other radio technologies)
+
+#### Story 9: region specific data sovereignty
+As a developer, I want to ensure that my devices connect to the specific regional endpoints to ensure compliance with data sovereignty.
+
+### Notes/Constraints/Caveats
+
+Traceability of device spec changes over time becomes increasingly important with templating,
```

Comment:
I think this is an important point, but also goes beyond the scope of this enhancement.  I think we'd want a separate enhancement describing an audit mechanism and detailing what events to audit.